ProjectSource/app-library/src/test/java/com/kayako/sdk/android/k5/messenger/messagelistpage/MessageListContainerFragmentTest.java

```diff
@@ -0,0 +1,403 @@
+package com.kayako.sdk.android.k5.messenger.messagelistpage;
+
+import static org.hamcrest.core.Is.is;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertThat;
+
+import android.app.Activity;
+import android.content.Context;
+import android.content.Intent;
+import android.content.res.Resources;
+import android.os.Bundle;
+import android.support.v4.app.Fragment;
+import android.support.v4.app.FragmentActivity;
+import android.view.Gravity;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.Toast;
+import com.kayako.sdk.android.k5.R;
+import com.kayako.sdk.android.k5.activities.KayakoAttachmentPreviewActivity;
+import com.kayako.sdk.android.k5.activities.KayakoSelectConversationActivity;
+import com.kayako.sdk.android.k5.common.adapter.BaseListItem;
+import com.kayako.sdk.android.k5.common.utils.file.FileAttachmentUtil;
+import com.kayako.sdk.android.k5.core.HelpCenterPref;
+import com.kayako.sdk.android.k5.core.Kayako;
+import com.kayako.sdk.android.k5.core.MessengerPref;
+import com.kayako.sdk.android.k5.messenger.replyboxview.ReplyBoxContract;
+import com.kayako.sdk.android.k5.messenger.toolbarview.MessengerToolbarContract;
+import java.util.ArrayList;
+import java.util.List;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+import org.powermock.reflect.Whitebox;
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({
+    MessageListContainerFragment.class,
+    HelpCenterPref.class,
+    Kayako.class,
+    MessengerPref.class,
+    MessageListContainerFactory.class,
+    FragmentActivity.class,
+    Intent.class,
+    Bundle.class,
+    FileAttachmentUtil.class,
+    Fragment.class,
+    Toast.class,
+    Gravity.class
+})
+public class MessageListContainerFragmentTest {
+
+  private MessageListContainerFragment messageListContainerFragment;
+
+  private static final boolean SAVED_INSTANCE_STATE = false;
+  private static final int RESULT_CODE_OK = 1;
+  private static final String TOAST_MESSAGE = "HELLO";
+  private static final String HINT_MESSAGE = "TEST MESSAGE";
+
+  @Mock
+  private Bundle bundle;
+
+  @Mock
+  private Context context;
+
+  @Mock
+  private MessageListContainerContract.Data contractData;
+
+  @Mock
+  private LayoutInflater inflater;
+
+  @Mock
+  private ViewGroup viewGroup;
+
+  @Mock
+  private View view;
+
+  @Mock
+  private FragmentActivity fragmentActivity;
+
+  @Mock
+  private Intent intent;
+
+  @Mock
+  private Resources resources;
+
+  @Rule
+  private ExpectedException thrown = ExpectedException.none();
```

Comment:
❗️ This snippet of code contains an instance variable assigned when declared, but not `final`. Consider marking this instance variable as `final` so, it should become a **final** instance variable. This way the code avoids intentionally/accidentally re-assignment of this instance variable and sustain the best OO practices.  

**[More details](https://confluence.devfactory.com/x/OKXNF)**
          
## The solution in this case
```java
@Rule 
private final ExpectedException thrown = ExpectedException.none();
``` 

<a id="66b39090-4f1d-11e8-b712-a55fdd633c3d" href="none" codereview-uuid-reserved-tag />